be/src/util/system_metrics.cpp

```diff
@@ -292,6 +297,33 @@ void SystemMetrics::_install_memory_metrics(MetricRegistry* registry) {
     registry->register_metric("datacache_mem_bytes", &_memory_metrics->datacache_mem_bytes);
 }
 
+void SystemMetrics::_update_datacache_mem_tracker() {
+    // update datacache mem_tracker
+    int64_t datacache_mem_bytes = 0;
+    auto* datacache_mem_tracker = GlobalEnv::GetInstance()->datacache_mem_tracker();
+    if (datacache_mem_tracker) {
+        BlockCache* block_cache = BlockCache::instance();
+        if (block_cache->is_initialized()) {
+            auto datacache_metrics = block_cache->cache_metrics();
+            datacache_mem_bytes = datacache_metrics.mem_used_bytes + datacache_metrics.meta_used_bytes;
+        }
+#ifdef USE_STAROS
+        if (!config::datacache_unified_instance_enable) {
+            datacache_mem_bytes += staros::starlet::fslib::star_cache_get_memory_usage();
+        }
+#endif
+        datacache_mem_tracker->set(datacache_mem_bytes);
+    }
+}
+
+void SystemMetrics::_update_pagecache_mem_tracker() {
+    auto* pagecache_mem_tracker = GlobalEnv::GetInstance()->page_cache_mem_tracker();
+    auto* page_cache = StoragePageCache::instance();
+    if (pagecache_mem_tracker && page_cache) {
+        pagecache_mem_tracker->set(page_cache->memory_usage());
```

Comment:
So the `pagecache_mem_tracker` will only trace the memory used by cache data and ignore the metadata memory usage?

Reply:
Does the Metadata memory mentioned here refer to the memory occupied by the key, or the memory occupied by the hash table itself?